client/state/data-layer/wpcom/read/streams/index.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * External dependencies
+ */
+import { find, identity, random } from 'lodash';
+import { translate } from 'i18n-calypso';
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import warn from 'lib/warn';
+import {
+	isRequestInflight,
+	markRequestInflight,
+	completeRequest
+} from 'lib/inflight';
+import {
+	READER_STREAMS_PAGE_REQUEST
+} from 'state/action-types';
+import { receivePage } from 'state/reader/streams/actions';
+import { errorNotice } from 'state/notices/actions';
+
+/**
+ * Pull the suffix off of a stream ID
+ *
+ * A stream id is a : delimited string, with the form
+ * {prefix}:{suffix}
+ *
+ * Prefix cannot contain a colon, while the suffix may.
+ *
+ * A colon is not required.
+ *
+ * Valid IDs look like
+ * `following`
+ * `site:1234`
+ * `search:a:value` ( prefix is `search`, suffix is `a:value` )
+ *
+ * @param  {String} streamId The stream ID to break apart
+ * @return {String}          The stream ID suffix
+ */
+function streamIdSuffix( streamId ) {
+	return streamId.substring( streamId.indexOf( ':' ) + 1 );
+}
+
+/**
+ * A Higher Order Function that generates a function that replaces a named token with the stream suffix
+ *
+ * Usage:
+ *   `replacePathTokenWithSuffix( 'name' )( '/a/path/:name', 'name:ben' )`
+ * @param  {String} token The named token to replace
+ * @return {Function}       A function that takes a path and an action with a `streamId` property
+ */
+function replacePathTokenWithSuffix( token ) {
```

Comment:
I don't love this.  I feel like we should figure out how to use either es6 template literals or lodash's `template`.

e.g:
```jsx
{
    match: /^feed:/,
    path: template( '/read/feed/<%= feed >/posts' ),
},

....

function requestPage( ... ) {
  ....
  const actualPath = api.path( streamIdSuffix( streamId ) )
} 
```